### PR TITLE
Allow to enable Proof Key for Code Exachange (PKCE)

### PR DESCRIPTION
**Pull request by thalionath**
**Describe the PR**
Wires usePkceWithAuthorizationCodeGrant OAuth2 option of the Swagger UI to the options interface

**Relation issue**
Closes https://github.com/swaggo/gin-swagger/issues/243

**Additional context**
Allow library users to set the usePkceWithAuthorizationCodeGrant OAuth2 config option of the SwaggerUI
https://swagger.io/docs/open-source-tools/swagger-ui/usage/oauth2/

### CODE REVIEW

File: swagger.go [hunk 1]
```diff
@@ -106,17 +109,26 @@ func Oauth2DefaultClientID(oauth2DefaultClientID string) func(*Config) {
 	}
 }
 
+// UsePkceWithAuthorizationCodeGrant enables Proof Key for Code Exchange.
+// Only applies to accessCode (Authorization Code) flows.
+func UsePkceWithAuthorizationCodeGrant(usePkce bool) func(*Config) {
```
Comment (ubogdan) [inline swagger.go:114]: Please rename the option to `Oauth2UsePkce` and you can leave the details the function doc.

File: swagger.go [hunk 2]
```diff
@@ -270,12 +282,10 @@ window.onload = function() {
...
 
-  const defaultClientId = "{{.Oauth2DefaultClientID}}";
-  if (defaultClientId) {
-    ui.initOAuth({
-      clientId: defaultClientId
-    })
-  }
+  ui.initOAuth({
+    clientId: {{.Oauth2DefaultClientID}},
+    usePkceWithAuthorizationCodeGrant: {{.UsePkceWithAuthorizationCodeGrant}}
```
Comment (ubogdan) [inline swagger.go:287]: I hope this is backward compatible.

**Review by ubogdan — CHANGES_REQUESTED** [review]

File: swagger.go (hunk 2 above)
Comment (thalionath) [reply swagger.go]: No, it was not indeed. I changed it back. Calling Oauth2DefaultClientID will enable oauth2 and take the pkce property into account only then.

**Review by ubogdan — APPROVED** [review]

**Comment by ubogdan** [comment]
@thalionath Thanks for your contribution

**Pull request merged**